video/out/w32_common.c

```diff
@@ -466,6 +464,59 @@ static void wakeup_gui_thread(void *ctx)
     }
 }
 
+struct get_monitor_data {
+    int i;
+    int target;
+    HMONITOR mon;
+};
+
+static BOOL CALLBACK get_monitor_proc(HMONITOR mon, HDC dc, LPRECT r, LPARAM p)
+{
+    struct get_monitor_data *data = (struct get_monitor_data*)p;
+
+    if (data->i == data->target) {
+        data->mon = mon;
+        return FALSE;
+    }
+    data->i++;
+    return TRUE;
+}
+
+static HMONITOR get_monitor_from_id(int id)
+{
+    struct get_monitor_data data = { .target = id };
+    EnumDisplayMonitors(NULL, NULL, get_monitor_proc, (LPARAM)&data);
+    return data.mon;
+}
+
+static HMONITOR get_monitor_from_window(HWND hwnd)
```

Comment:
This seems redundant (and get_primary_monitor() does as well.)

I'm all for utility functions like get_monitor_from_id(), but get_monitor_from_id() wraps quite a bit of complexity and these are just forwarders for MonitorFromWindow().